python_modules/libraries/dagster-components/dagster_components/core/dsl_schema.py

```diff
@@ -20,21 +20,34 @@ class OpSpecBaseModel(BaseModel):
     tags: Optional[Dict[str, str]] = None
 
 
-class AutomationConditionModel(BaseModel):
-    type: str
-    params: Mapping[str, Any] = {}
+class AssetAttributesModel(BaseModel):
+    key: Optional[str] = None
+    deps: Sequence[str] = []
+    description: Optional[str] = None
+    metadata: Union[str, Mapping[str, Any]] = {}
+    group_name: Optional[str] = None
+    skippable: bool = False
+    code_version: Optional[str] = None
+    owners: Sequence[str] = []
+    tags: Union[str, Mapping[str, str]] = {}
+    automation_condition: Optional[Union[str, AutomationCondition]] = RenderingScope(
+        Field(None), required_scope={"automation_condition"}
+    )
```

Comment:
Technically, this field should actually just be of type Optional[str].

I want to do a followup here where we do something like:

```python

automation_condition: Optional[str] = RenderedField(Optional[AutomationCondition], required_scope={"automation_condition"})
```

where we can have separate type checking for the rendered value

Reply:
this would be a great followup.

Leave comment to this effect explaining the next steps.